manifests/opsmanager.pp

```diff
@@ -23,30 +44,77 @@
   Boolean $ssl                                   = false,
   Boolean $ignore_ui_setup                       = true,
   #optional settings
-  Optional[String[1]] $ca_file                   = $mongodb::opsmanager::params::ca_file,
-  Optional[String[1]] $pem_key_file              = $mongodb::opsmanager::params::pem_key_file,
-  Optional[String[1]] $pem_key_password          = $mongodb::opsmanager::params::pem_key_password,
+  Optional[String[1]] $ca_file                   = undef,
+  Optional[String[1]] $pem_key_file              = undef,
+  Optional[String[1]] $pem_key_password          = undef,
   Optional[String[1]] $user_svc_class            = undef, # Default: com.xgen.svc.mms.svc.user.UserSvcDb External Source: com.xgen.svc.mms.svc.user.UserSvcCrowd or Internal Database: com.xgen.svc.mms.svc.user.UserSvcDb
   Optional[Integer] $snapshot_interval           = undef, # Default: 24
   Optional[Integer] $snapshot_interval_retention = undef, # Default: 2
   Optional[Integer] $snapshot_daily_retention    = undef, # Default: 0
   Optional[Integer] $snapshot_weekly_retention   = undef, # Default: 2
   Optional[Integer] $snapshot_monthly_retention  = undef, # Default: 1
   Optional[Integer] $versions_directory          = undef, # Linux default: /opt/mongodb/mms/mongodb-releases/
+) {
+
+  case $package_ensure {
+    'absent': {
+      $my_package_ensure = 'absent'
+      $file_ensure       = 'absent'
+    }
+    default:  {
+      $my_package_ensure = $package_ensure
+      $file_ensure       = 'present'
+    }
+  }
+
+  if versioncmp(fact('puppetversion'),'5.4.0') < 0 {
```

Comment:
kind of related question. Which puppet 5 version do we want to support after we drop puppet4 in december? latest puppet 5 minor version?

Reply:
I think we should stick with the requirements at ```metadata.json```.

Reply:
Since metadata.json now requires `>= 5.5.8` I've dropped this.